hazelcast/include/hazelcast/client/IFuture.h

```diff
@@ -0,0 +1,135 @@
+ * limitations under the License.
+ */
+#ifndef HAZELCAST_CLIENT_IFUTURE_H_
+#define HAZELCAST_CLIENT_IFUTURE_H_
+
+#include <boost/shared_ptr.hpp>
+
+#include "hazelcast/client/exception/InterruptedException.h"
+#include "hazelcast/util/concurrent/TimeUnit.h"
+
+using namespace hazelcast::client;
+using namespace hazelcast::util::concurrent;
+
+#if  defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#pragma warning(push)
+#pragma warning(disable: 4251) //for dll export	
+#endif
+
+namespace hazelcast {
+    namespace client {
+
+        /**
+         * A {@code Future} represents the result of an asynchronous
+         * computation.  Methods are provided to check if the computation is
+         * complete, to wait for its completion, and to retrieve the result of
+         * the computation.  The result can only be retrieved using method
+         * {@code get} when the computation has completed, blocking if
+         * necessary until it is ready.  Cancellation is performed by the
+         * {@code cancel} method.  Additional methods are provided to
+         * determine if the task completed normally or was cancelled. Once a
+         * computation has completed, the computation cannot be cancelled.
+         *
+         *
+         * @param <V> The result type returned by this Future's {@code get} method
+         */
+        template<typename V>
+        class IFuture {
+        public:
+            virtual ~IFuture() {
+            }
+
+            /**
+             * Attempts to cancel execution of this task.  This attempt will
+             * fail if the task has already completed, has already been cancelled,
+             * or could not be cancelled for some other reason. If successful,
+             * and this task has not started when {@code cancel} is called,
+             * this task should never run.  If the task has already started,
+             * then the {@code mayInterruptIfRunning} parameter determines
+             * whether the thread executing this task should be interrupted in
+             * an attempt to stop the task.
+             *
+             * <p>After this method returns, subsequent calls to {@link #isDone} will
+             * always return {@code true}.  Subsequent calls to {@link #isCancelled}
+             * will always return {@code true} if this method returned {@code true}.
+             *
+             * @param mayInterruptIfRunning {@code true} if the thread executing this
+             * task should be interrupted; otherwise, in-progress tasks are allowed
+             * to complete
+             * @return {@code false} if the task could not be cancelled,
+             * typically because it has already completed normally;
+             * {@code true} otherwise
+             */
+            virtual bool cancel(bool mayInterruptIfRunning) = 0;
+
+            /**
+             * Returns {@code true} if this task was cancelled before it completed
```

Comment:
If the return parameter explanation is enough for the function why is there a duplication? Both return type and function have the same sentence.

Reply:
It is from java.util.concurrent.Future doc. I believe the upper description is as a general description of the method while the return statement and parameters would be the details. Yes there seems to be duplication but since Java did it this way so I would go with it.